langchain4j-core/src/main/java/dev/langchain4j/model/embedding/DimensionAwareEmbeddingModel.java

```diff
@@ -0,0 +1,36 @@
+package dev.langchain4j.model.embedding;
+
+import dev.langchain4j.data.embedding.Embedding;
+
+import java.util.Optional;
+
+/**
+ * A dimension aware embedding model
+ */
+public abstract class DimensionAwareEmbeddingModel implements EmbeddingModel {
+
+    /**
+     * dimension of embedding
+     */
+    protected Integer dimension;
+
+    /**
+     * Returns the dimension of known {@link Embedding} produced by this embedding model. if it's unknown, return null
```

Comment:
```suggestion
     * When known (e.g., can be derived from the model name), returns the dimension of the {@link Embedding} produced by this embedding model. Otherwise, it returns {@code null}.
```